packages/docusaurus-plugin-content-docs/src/globalData.ts

```diff
@@ -20,13 +24,44 @@ export function toGlobalDataDoc(doc: DocMetadata): GlobalDoc {
   };
 }
 
+export function toGlobalSidebar(
+  sidebarId: string,
+  sidebar: Sidebar,
+  version: LoadedVersion,
+): GlobalSidebar {
+  const sidebarFirstDocId = getFirstDocIdOfSidebar(sidebar);
+  let sidebarFirstDocPermalink;
+  if (!sidebarFirstDocId.isAutoGeneratedIndex) {
+    sidebarFirstDocPermalink = version.docs.find(
+      (doc) => doc.id === sidebarFirstDocId.firstDocId,
+    )?.permalink;
+  } else {
+    sidebarFirstDocPermalink = normalizeUrl([
+      version.versionPath,
+      sidebarFirstDocId.firstDocId,
+    ]);
+  }
+  return {
+    link: {
+      label: sidebarId,
+      path: sidebarFirstDocPermalink ?? sidebarFirstDocId.firstDocId,
+    },
+  };
+}
+
 export function toGlobalDataVersion(version: LoadedVersion): GlobalVersion {
+  const globalSidebarRecord: Record<string, GlobalSidebar> = {};
+  Object.entries(version.sidebars).forEach(([sidebarId, sidebar]) => {
```

Comment:
Pretty sure we can have a simple `fromEntries` here so we don't have side-effects.